framework/behaviors/AttributeBehavior.php

```diff
@@ -118,6 +123,10 @@ public function evaluateAttributes($event)
             foreach ($attributes as $attribute) {
                 // ignore attribute names which are not string (e.g. when set by TimestampBehavior::updatedAtAttribute)
                 if (is_string($attribute)) {
+                    // skip attribute update when it is not empty and the corresponding mode is enabled
+                    if ($this->skipUpdateOnFilled && !empty($this->owner->$attribute)) {
```

Comment:
@dynasource I refactored the code, what would you say about `preserveExistingValues`?

```php
if ($this->preserveExistingValues && !empty($this->owner->$attribute)) {
    continue;
}
$this->owner->$attribute = $value;
```

`empty` is used here since we want to skip the attribute evaluation both when it's an empty string and `null`.

No event name check is needed here since we want to skip non-empty attributes evaluation while insert too.